rfcs/text/0000-dss-deletion-daemon.md

```diff
@@ -0,0 +1,620 @@
+   
+   A notification is sent out to subscribers of datastore deletions when a new bundle tombstone is created, so they can 
+   update their index appropriately. If the the bundle was logically deleted and later physically deleted a second
+   identical notification will be sent to subscribers. The subscribers must be able to handle these notifications 
+   idempotently.
+   
+   A **physical deletion** request of a bundles results in the bundle associated files being added to 
+   the deletion queue for regularly schedule **Physical Deletion**. A **physical deletion** request of a file result in 
+   that file being added to the to deletion queue
+
+5) On a regular schedule the deletion queue is processed by code running within a secure boundary. An administrator 
+   may manually invoke the deletion process or wait for it run at it's scheduled time once daily.
+   The deletion daemon is responsible for performing **physical deletes** of data from the deletion queue. 
+   A physical delete makes all files associated with a bundle or bundle version inaccessible. This will cause direct
+   urls to a deleted file to fail.
+   
+   A log entry is produced when ever the deletion daemon is run. It contains information on what bundles and files
+   have been logically deleted, the reason, the administrator who executed the deletion, and info about the deletion markers. 
+   The same information is stored in the **Deletion Table**. The deletion table tracks all bundles and files currently
+   pending permanent deletion. The Deletion Table contains enough information to reverse Physical Deletes before 
+   the data is Permanently deleted.
+   
+   A daily digest of bundles and files that are scheduled for deletion are sent to DCP administrator 24 hours prior to the data
+   being permanently deleted.
+
+6) Data that has been physical deleted remain in the bucket for 7 days before being permanently deleted from the bucket.
+   This grace period allows the deletion to be reversed within that time period. After the grace period the data will 
+   be permanently deleted.
+ 
+### Tombstones
+
+Tombstones are markers places in the DSS to indicate previously existing data has been removed. Tombstones exist for
+files and bundles stored in the DSS. Tombstones come in two varieties, versioned and unversioned tombstones. A 
+version tombstone marks the deletion of a specific version of a file or bundle. This prevents that same version of data 
+from being reuploaded or retrieved. An unversioned tombstone does not specify a version and prevent a file or bundle 
+using the same UUID from being reuploaded or retrieved. Whether a tombstone is version or unversioned is determined when
+during a deletion request. If the version parameter is present in the request, then a versioned tombstone is created. If
+no version is specified then an unversion tombstone is created. A versioned tombstone is created for all bundles and 
```

Comment:
This is a little misleading. If no version is specified, actually an _unversioned_ tombstone is created that is applied to all the bundles with that UUID.